### PR TITLE
Bind address page to state and add location change modal

### PR DESCRIPTION
**Pull request by ranzwertig**
# Definition of Done

- [ ] The code does at least what is defined in the ticket and does not affect any other functionality
- [ ] The PR has a meaningful title
- [ ] There is a link to the issue-id
- [ ] There are no 'hacks' or shortcuts refactoring is preferred
- [ ] The happy case of the app was tested at least once manually

### CODE REVIEW

**Comment by webholics** [comment]
Der build failed.

File: src/app/index.js [hunk 1]
```diff
@@ -59,3 +60,7 @@ if (process.env.NODE_ENV !== 'production') {
     })
   }
 }
+
+global.dev = {
```
Comment (webholics) [inline src/app/index.js]: willst du das wirklich drin lassen?

File: src/app/action-next/core.js [hunk 2]
```diff
@@ -60,3 +64,13 @@ export const updateShippings = (payload: Array<Shipping>): UpdateShippingsAction
   type: 'CORE.UPDATE_SHIPPINGS',
   payload
 })
+
+export const saveUser = (user: User): SaveUserAction => ({
+  type: 'CORE.SAVE_USER',
+  payload: user
+})
+
+export const updateUser = (payload: {userId: UserId}): UpdateUserAction => ({
```
Comment (webholics) [inline src/app/action-next/core.js]: hier wie besprochen am besten sogar `userReceived` nennen

File: src/app/container-next/address-page.js [hunk 3]
```diff
@@ -322,70 +338,101 @@ const mapStateToProps = state => ({
...
+        'billingAddress.countryCode',
+        stateProps.shippingAddress.countryCode
       )
     }
   }
-}
+})
 
 const enhance = compose(
   // TODO: enable the guard again so thet the page can be
```
Comment (webholics) [inline src/app/container-next/address-page.js]: den guard wieder aktivieren

File: src/app/container-next/modal/pick-location.js [hunk 4]
```diff
@@ -49,8 +57,18 @@ const PickLocationModal = ({
...
+      headline={meta.showWarning ? headlineWarning : headline}
+      buttons={compact(buttons)}
+      closeable={false}
+    >
+      {meta.showWarning ? (
+        <Paragraph>
+          If you change your country or currency you have to reconfigure all models.
+        </Paragraph>
+      ) : (
+        <Paragraph>We need your address and currency to calculate the shipping prices</Paragraph>
```
Comment (webholics) [inline src/app/container-next/modal/pick-location.js]: hier fehlt ein Punkt am ende

File: src/app/container/success-page.js [hunk 5]
```diff
@@ -14,7 +14,6 @@ import AppLayout from './app-layout'
 import {getProviderName} from '../lib/provider-selector'
```
Comment (webholics) [inline src/app/container/success-page.js]: warum hast du in der file was geändert?

Comment (ranzwertig) [reply src/app/container/success-page.js]: ui das war wohl iwi ein merge aus dem master

**Pull request merged**